src/store/sagas/requests/adminQueryUsers.js

```diff
@@ -0,0 +1,16 @@
+import firebase from '../../../utils/Firebase'
+
+const db = firebase.firestore()
+
+export const getUnapprovedUsers = async () => {
+    let data = await db.collection('users').get()
+    let unapprovedUsers = []
+    for (let user of data.docs) {
+        let privateInfo = await db.doc(`users/${user.id}/private/info`).get()
+        if (!privateInfo.data().isApproved) {
+            let userData = await db.doc(`newUser/${privateInfo.data().email}`).get()
+             unapprovedUsers.push(userData.data())
```

Comment:
You have to get `users` that have `isApproved` set to `false`, not all the `users`. There is a `where` method in firebase which you can use to filter document gets. Imagine having only 1 unapproved user and you'd have to get over 500 documents just for that 1 user. Pay Per document read

Reply:
The `user` collection itself doesn't have a `isApproved` field.

Reply:
???

Reply:
Ooh thanks, just understamding what you meant here

Reply:
I'm having issue trying to chain the query

Reply:
What's the issue?

Reply:
https://firebase.google.com/docs/firestore/query-data/queries#collection-group-query

Reply:
Don't you think the `isApproved`  field should be inside the `users` collection directly and not in a subcollection.

Reply:
all the data that is inside the `private/info` document are data that the user cannot edit. the data that is directly in the `users` document is the data the user can edit.

We wouldn't want the user to be able to edit his `wallet`, `approval` status, `uid`, `role` etc...